src/main/java/org/wildfly/naming/client/remote/RemoteServerTransport.java

```diff
@@ -76,8 +79,13 @@ final class RemoteServerTransport implements RemoteTransport {
         this.messageTracker = messageTracker;
         this.localContext = localContext;
         this.configuration = new MarshallingConfiguration();
-        configuration.setVersion(version == 2 ? 4 : 2);
+        configuration.setVersion(version >= 2 ? 4 : 2);
```

Comment:
Yes, above 'configuration.setVersion()' statement is correct because it is configuring marshalling version and not JNDI client protocol version @chengfang . Also note that JNDI client protocol version can be configured via protocol itself and so it is not guaranteed the version is 3 or above.